CMakePresets.json

```diff
@@ -52,16 +52,30 @@
       }
     },
     {
+      "name": "win-vs",
+      "displayName": "win-vs",
+      "description": "Default build options for a windows visual studio conda build",
+      "binaryDir": "${sourceDir}/build",
+      "generator": "Visual Studio 17 2022",
```

Comment:
I think at the moment this doesn't hit us as we don't use super advance features in C++ but Conda are using VS2019 (https://conda-forge.org/docs/user/announcements.html#moving-to-visual-studio-toolchain-vc142) and are therefore using the `vc141` toolset. For 2022 can we also set the toolset version to `vc141` so that we can use the IDE but match the toolset for condaforge to ensure compatability: https://cmake.org/cmake/help/latest/variable/MSVC_TOOLSET_VERSION.html

Reply:
Perhaps we should then also get developers to move to 2022 as a default?

Reply:
As discussed I'll try and force the toolset version in cmake. I guess I should force to `vc142` rather than `vc141`?

Re getting developers to move to VS2022 - yes that's my intention. I've updated the "Getting Started with Windows..." doc page to say developers should install 2022. I've been less definite on using the VS generator vs ninja. The instructions just explain how to set up either.

Reply:
I've just pushed the changes to force the v142 toolset. A few comments on this:
- I've put the change in the CMakePresets.json file because it's convenient but also because I had to - the CMake documentation suggests setting the toolset in your CMake project code is unsupported. I did manage to get it to work that way by setting the CMake variable right at the start before the initial `project(Mantid)` line in the main CMakeLists.txt file but it seems a bit risky and also a bit odd putting it there
- the v142 toolset isn't installed with VS2022 by default so I've added a note to the documentation about that
 - the change only affects the Visual Studio generator. The ninja generator doesn't support the toolset version being specified so still some risk of v143 being used there
- the Visual Studio 2022 UI highlights the fact we're running the old toolset in quite a verbose way but not a big deal I think:

![image](https://user-images.githubusercontent.com/56295817/217585936-ba811544-c6e2-498e-9969-d061c05ac498.png)

Reply:
OK thanks. We're getting the compiler tools into the `PATH` at the moment for ninja users by advising developers to run the `cmake` commands from the Visual Studio command prompt. The Visual Studio 2022 command prompt has the v143 toolset in its path though so that's the problem we'd have to fix